std/assembly/internal/string.ts

```diff
@@ -10,10 +10,10 @@ export const MAX_LENGTH = (<i32>MAX_SIZE_32 - HEADER_SIZE) >>> 1;
 
 function __gc(ref: usize): void {}
 
-export function allocateUnsafe(length: i32): String {
+export function allocateUnsafe(length: i32, skipGC: bool = false): String {
   assert(length > 0 && length <= MAX_LENGTH);
   var buffer: usize;
-  if (isManaged<String>()) {
+  if (isManaged<String>() && !skipGC) {
```

Comment:
Not sure about this. `isManaged` can be precomputed so that the entire branch can be DCE'ed, but with `!skipGC` this doesn't work anymore.

Reply:
See this [fiddle](https://webassembly.studio/?f=vj4mfjap7s). It still precompute

Reply:
Can you make this two nested ifs, just to be sure?

Reply:
Yeah. Good idea